plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotRestoreTests.java

```diff
@@ -70,20 +70,16 @@ private String getRepositoryPath() {
     }
 
     public static String getContainerName() {
-        String testName = "snapshot-itest-".concat(RandomizedTest.getContext().getRunnerSeedAsString().toLowerCase(Locale.ROOT));
+        /* Have a different name per test so that there is no possible race condition. As the long can be negative,
+         * there mustn't be an hyphen between the 2 concat numbers
+         * (can't have 2 consecutives hypens on Azure containers)
+         */
+        String testName = "snapshot-itest-"
+            .concat(RandomizedTest.getContext().getRunnerSeedAsString().toLowerCase(Locale.ROOT))
+            .concat(new Long(RandomizedTest.getContext().getRandom().nextLong()).toString());
```

Comment:
This can simply be `randomLong()`. However, I don't think this will work because  the cleanup in `wipeAzureRepositories()` would then not know about the generated name. If these need to be unique, then this method needs to stash the generated name so that the repository can be cleaned up after the test.